src/lamp_py/bus_performance_manager/vehicle_events.py

```diff
@@ -0,0 +1,133 @@
+    """
+    get new realtime files from the vehicle position and transit master s3
+    locations that have been populated since the last time vehicle events were
+    processed.
+    """
+    latest_event_file = get_last_modified_object(
+        bucket_name=RemoteFileLocations.bus_events.bucket_name,
+        file_prefix=RemoteFileLocations.bus_events.file_prefix,
+        version="1.0",
+    )
+
+    if latest_event_file:
+        cutoff = latest_event_file["last_modified"] - timedelta(hours=1)
+        vp_files = file_list_after(
+            bucket_name=RemoteFileLocations.vehicle_positions.bucket_name,
+            file_prefix=RemoteFileLocations.vehicle_positions.file_prefix,
+            cutoff=cutoff,
+        )
+
+        tm_files = file_list_after(
+            bucket_name=RemoteFileLocations.tm_stop_crossing.bucket_name,
+            file_prefix=RemoteFileLocations.tm_stop_crossing.file_prefix,
+            cutoff=cutoff,
+        )
+
+    else:
+        vp_files = file_list_from_s3(
+            bucket_name=RemoteFileLocations.vehicle_positions.bucket_name,
+            file_prefix=RemoteFileLocations.vehicle_positions.file_prefix,
+        )
+
+        tm_files = file_list_from_s3(
+            bucket_name=RemoteFileLocations.tm_stop_crossing.bucket_name,
+            file_prefix=RemoteFileLocations.tm_stop_crossing.file_prefix,
+        )
+
+    return vp_files, tm_files
+
+
+def files_per_service_date(
+    vp_files: List[str], tm_files: List[str]
+) -> Dict[date, BusInputFiles]:
+    """
+    take a list of input files and bucket them based on the service date they
+    cover.
+    """
+
+    service_dates: Dict[date, BusInputFiles] = {}
+
+    def add_vp_file(service_date: date, vp_file: str) -> None:
+        """helper to add vp file to service dates"""
+        if service_date not in service_dates:
+            service_dates[service_date] = BusInputFiles(
+                service_date=service_date
+            )
+
+        service_dates[service_date].vp_files.append(vp_file)
+
+    def add_tm_file(service_date: date, tm_file: str) -> None:
+        """helper to add tm file to service dates"""
+        if service_date not in service_dates:
+            service_dates[service_date] = BusInputFiles(
+                service_date=service_date
+            )
+
+        service_dates[service_date].tm_files.append(tm_file)
+
+    # parse vp filepaths to infer service dates from the partition paths
+    for vp_file in vp_files:
+        file_datetime = get_datetime_from_partition_path(vp_file)
+
+        # if the partition is hourly (for older files), consider the hour of
+        # the file to determine the service date from the file datetime
+        if "hour" in vp_file:
+            if file_datetime.hour < 6:
+                yesterday = (file_datetime - timedelta(days=1)).date()
+                add_vp_file(yesterday, vp_file)
+            if file_datetime.hour > 3:
+                add_vp_file(file_datetime.date(), vp_file)
```

Comment:
`file_datetime` here should be a UTC datetime, so I don't think these hour conditionals would be applicable.